src/ServerRequestFilter/FilterUsingXForwardedHeaders.php

```diff
@@ -76,7 +77,8 @@ public function __invoke(ServerRequestInterface $request): ServerRequestInterfac
 
             switch ($headerName) {
                 case self::HEADER_HOST:
-                    $uri = $uri->withHost($header);
+                    [$host] = UriFactory::marshalHostAndPortFromHeader($header);
+                    $uri    = $uri->withHost($host);
```

Comment:
Interesting, does `withHost()` not reject invalid hostnames containing a `:`?

Reply:
Unfortunately not.
https://github.com/laminas/laminas-diactoros/blob/b2355dc58253c480aa25ba085845a5aabf84955d/src/Uri.php#L296-L315